### PR TITLE
add SignTransactions gRPC.

### LINKED ISSUE #1110: Add SignTransactions gRPC call
Idea is to have the same prototype as the existing `SignTransaction`, but supporting the signing of multiple transactions. This is better than calling `SignTransaction` multiple times as the wallet is unlocked only once when you need to sign multiple transactions at the same time.

Initial idea for the endpoint:

```protobuf3
[...]
   rpc SignTransactions (SignTransactionsRequest) returns (SignTransactionsResponse);
[...]
message SignTransactionsRequest {
	message AdditionalScript {
		bytes transaction_hash = 1;
		uint32 output_index = 2;
		int32 tree = 3;
		bytes pk_script = 4;
	}
        message UnsignedTransaction {
	        bytes serialized_transaction = 1;
        	repeated uint32 input_indexes = 2;                
        }

	bytes passphrase = 1;
        repeated UnsignedTransaction transactions = 2;
	repeated AdditionalScript additional_scripts = 3;
}
message SignTransactionsResponse {
        message SignedTransaction {
               bytes transaction = 1;
               repeated uint32 unsigned_input_indexes = 2;
        }
	repeated SignedTransactions transactions = 1;
}
```

### PR DESCRIPTION
**Pull request by dnldd**
resolves #1110.

### CODE REVIEW

File: rpc/api.proto [hunk 1]
```diff
@@ -10,8 +10,8 @@ message VersionRequest {}
 message VersionResponse {
 	string version_string = 1;
 	uint32 major = 2;
-	uint32 minor = 3;
-	uint32 patch = 4;
+	uint32 minor = 4;
+	uint32 patch = 0;
```
Comment (matheusd) [inline rpc/api.proto]: This doesn't need to change.

File: rpc/documentation/api.md [hunk 2]
```diff
@@ -1371,7 +1372,7 @@ transaction using a wallet private keys.
   input that is missing an input script.
 
 - `repeated AdditionalScript additional_scripts`: Additional output scripts of
-  previous outputs spent by the transaction that the wallet may not be aware of.
+  previous outputs spent by the transactions that the wallet may not be aware of.
```
Comment (matheusd) [inline rpc/documentation/api.md]: "transaction" was correct here. This "transaction" refers to the transaction being signed, not to the transactions the wallet isn't aware of.

File: rpc/documentation/api.md [hunk 3]
```diff
@@ -1404,6 +1405,59 @@ transaction using a wallet private keys.
...
+	- `repeated uint32 unsigned_input_indexes`: The indexes of every input that an
+    input script could not be created for.
+
+  **Expected errors:**
+
+  - `InvalidArgument`: A serialized transaction can not be decoded.
+
+  - `Aborted`: The wallet database is closed.
+
+  - `InvalidArgument`: The private passphrase is incorrect.
```
Comment (matheusd) [inline rpc/documentation/api.md]: Sort the errors by type.

File: rpc/documentation/api.md [hunk 4]
```diff
@@ -1393,17 +1394,67 @@ transaction using a wallet private keys.
...
+
+  - `bytes passphrase`: The wallet's private passphrase.
+
+  - `repeated UnsignedTransaction unsigned_transaction`: - The unsigned transactions set.
+
+    **Nested message:** `UnsignedTransaction`
+
+    - `bytes serialized_transaction`: The transaction to add input signatures to.
+
+    - `repeated uint32 input_indexes`: The input indexes that signature scripts must
```
Comment (jrick) [inline rpc/documentation/api.md]: this was removed

File: rpc/documentation/api.md [hunk 5]
```diff
@@ -1393,17 +1394,67 @@ transaction using a wallet private keys.
...
+    be created for.  If there are no indexes, input scripts are created for every
+    input that is missing an input script.
+
+    - `repeated AdditionalScript additional_scripts`: Additional output scripts of
```
Comment (jrick) [inline rpc/documentation/api.md]: this is indented too far, it should be at the same indent level as `repeated UnsignedTransaction unsigned_transaction`.

**Pull request merged**